### PR TITLE
rendezvous uTP connection API

### PR DESCRIPTION
**Pull request by vinipsmaker**
Necessary to use the hole punched UDP sockets.

[Review on Reviewable](https://reviewable.io/reviews/maidsafe/crust/366)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @canndrew

(maidsafe_highfive has picked a reviewer for you, use r? to override)

**Comment by vinipsmaker** [comment]
Now that [commit https://github.com/maidsafe/crust/commit/e3bf5bd6ee96fb180741305587fbdada9642d3a9](https://github.com/maidsafe/crust/commit/e3bf5bd6ee96fb180741305587fbdada9642d3a9) is in place, PR is ready for review/merge.

File: src/service.rs [hunk 1]
```diff
@@ -642,9 +642,6 @@ mod test {
         assert!(runner2.join().is_ok());
     }
 
-    // Temporarily ignored this test, should be fixed once a pending PR
-    // to https://github.com/maidsafe/rust-utp/pull/5 gets merged.
-    #[ignore]
```
Comment (inetic) [inline src/service.rs]: :)

**Pull request merged**